src/ucp/dt/dt.c

```diff
@@ -46,3 +47,116 @@ size_t ucp_dt_pack(ucp_datatype_t datatype, void *dest, const void *src,
     state->offset += result_len;
     return result_len;
 }
+
+#define MAX_RKEY_BUFFER (256)
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_mem_type_unpack(ucp_worker_h worker, void *buffer, const void *recv_data,
+                    size_t recv_length, uct_memory_type_t mem_type)
+{
+    ucp_context_h context = worker->context;
+    ucp_ep_h ep = worker->mem_type_ep[mem_type];
+    ucp_md_map_t md_map = 0;
+    ucp_lane_index_t lane;
+    uct_md_h md;
+    unsigned md_index;
+    uct_mem_h memh[1];
+    ucs_status_t status;
+    char rkey_buffer[MAX_RKEY_BUFFER];
+    uct_rkey_bundle_t rkey_bundle;
+
+    if (recv_length == 0) {
+        return UCS_OK;
+    }
+
+    lane = ucp_ep_config(ep)->key.rma_lanes[0];
+    md_index = ucp_ep_md_index(ep, lane);
+    md = context->tl_mds[md_index].md;
+    ucs_assert(MAX_RKEY_BUFFER > context->tl_mds[md_index].attr.rkey_packed_size);
+
+    status = ucp_mem_rereg_mds(context, UCS_BIT(md_index), buffer,
+                               recv_length, UCT_MD_MEM_ACCESS_ALL, NULL, mem_type,
+                               NULL, memh, &md_map);
+    if (status != UCS_OK) {
+        goto err;
+    }
+
+    status = uct_md_mkey_pack(md, memh[0], rkey_buffer);
```

Comment:
maybe we should add UCT function like uct_md_memh_to_rkey() to avoid all this pack/unpack overhead

Reply:
you mean new intraface to MD component? if that is the case, will do that separate PR for API.